File: sample-projects/ruby/lib/de/sots/cellars_and_centaurs/domain/model/speed.rb
# frozen_string_literal: true

require_relative '../../version'

module De::Sots::CellarsAndCentaurs::Domain::Model
  class Speed
    attr_accessor :feet_per_round

    def initialize(feet_per_round)
      @feet_per_round = feet_per_round
    end

    def to_s
      "#{feet_per_round} ft."
    end
  end
end

File: sample-projects/csharp/src/De.Sots.CellarsAndCentaurs/Domain/Model/Speed.cs
namespace De.Sots.CellarsAndCentaurs.Domain.Model;

public sealed class Speed
{
    public int FeetPerRound { get; }

    public Speed(int feetPerRound)
    {
        FeetPerRound = feetPerRound;
    }

    public static Speed Of(int feetPerRound) => new(feetPerRound);

    public override string ToString() => $"{FeetPerRound} ft.";
}
